frontend/packages/dev-console/src/components/catalog/service/CatalogServiceProvider.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { CatalogExtensionHookOptions, CatalogItem } from '@console/plugin-sdk';
+import { ResolvedExtension } from '@console/dynamic-plugin-sdk/src/api/useResolvedExtensions';
+import { keywordCompare } from '../utils/utils';
+import useCatalogExtensions from '../hooks/useCatalogExtensions';
+import CatalogItemsLoader from './CatalogItemsLoader';
+
+export type CatalogService = {
+  type: string;
+  items: CatalogItem[];
+  itemsMap: { [key: string]: CatalogItem[] };
+  loaded: boolean;
+  loadError: any;
+  searchCatalog: (query: string) => CatalogItem[];
```

Comment:
Unsure if this is the best API for searching the catalog vs a simple util or something else. This feature isn't used yet in this PR. We can re-visit this when we get to implementing the search capabilities.